source/MaterialXGenArnold/ArnoldShaderGenerator.cpp

```diff
@@ -19,6 +19,9 @@ ArnoldShaderGenerator::ArnoldShaderGenerator()
                                       "volume_matte" };
 
     _syntax->registerReservedWords(reservedWords);
+
+    // Set colorspace argument for texture lookups
+    _tokenSubstitutions[T_FILE_EXTRA_ARGUMENTS] = ", \"colorspace\", file.colorspace";
```

Comment:
[Question] What is the first parameter?

Reply:
The argument name to indicate input colour  pace in Arnold.
If / when vanilla-OSL accepts this argument, the base OSL generator will use that name, and Arnold can be adjusted as needed to adopt this.
 Is that your question ?